caikit/runtime/train.py

```diff
@@ -141,57 +212,108 @@ def main() -> int:
+        exit(USER_ERROR_EXIT_CODE)
+    except ValueError as e:
+        message = f"Invalid value for one or more input parameters: {e}"
+        log.warning(
+            {
+                "log_code": "<COR65474760E>",
+                "message": message,
+                "stack_trace": traceback.format_exc(),
+            }
         )
-        raise
+    except Exception:
+        message = "Exception encountered when attempting to parse input parameters"
+        log.warning(
+            {
+                "log_code": "<COR17776549E>",
+                "message": message,
+                "stack_trace": traceback.format_exc(),
+            }
+        )
+        write_termination_log(message)
+        exit(USER_ERROR_EXIT_CODE)
 
-    # Convert datatypes to match the training API
-    training_service = ServicePackageFactory.get_service_package(
-        ServicePackageFactory.ServiceType.TRAINING,
-    )
-    train_rpcs = [
-        rpc
-        for rpc in training_service.caikit_rpcs.values()
-        if rpc.module_list == [module]
-    ]
-    error.value_check(
-        "<COR11978965E>",
-        len(train_rpcs) == 1,
-        "Unable to find a unique train signature",
-    )
-    package_name = get_service_package_name(ServicePackageFactory.ServiceType.TRAINING)
-    train_rpc_req = (
-        train_rpcs[0].create_request_data_model(package_name).get_proto_class()
-    )
-    request_proto = json_format.Parse(
-        json.dumps({"parameters": training_kwargs}),
-        train_rpc_req(),
-    )
-    req_kwargs = build_caikit_library_request_dict(
-        request_proto.parameters, module.TRAIN_SIGNATURE
-    )
-    train_kwargs.update(req_kwargs)
-    log.debug3("All train kwargs: %s", train_kwargs)
-
-    # Run the training
-    with alog.ContextTimer(
-        log.info,
-        "Finished training %s in: ",
-        args.model_name,
-    ):
-        future = train(module, wait=True, **train_kwargs)
-        info = future.get_info()
-        if info.status == TrainingStatus.COMPLETED:
-            log.info("Training finished successfully")
-            return 0
-        else:
-            log.error("Training finished unsuccessfully")
-            for err in info.errors or []:
-                log.error(err)
-            return 1
+    try:
+        # Run the training
+        with alog.ContextTimer(
+            log.info,
+            "Finished training %s in: ",
+            args.model_name,
+        ):
+            future = train(module, wait=True, **train_kwargs)
+            info = future.get_info()
+            if info.status == TrainingStatus.COMPLETED:
+                log.info("Training finished successfully")
+                return 0
+            else:
+                log.error("Training finished unsuccessfully")
+                for err in info.errors or []:
+                    log.error(err)
+                write_termination_log("Training finished unsuccessfully")
+                return INTERNAL_ERROR_EXIT_CODE
```

Comment:
Should this be `exit`?

Reply:
Yeah I was debating that. I think you're right, the `exit` makes more sense here.